python/tester/process_test.py

```diff
@@ -0,0 +1,146 @@
+from activator.visit import Visit
+
+logging.basicConfig(level=logging.INFO, stream=sys.stdout)
+_log = logging.getLogger(__name__)
+_log.setLevel(logging.DEBUG)
+
+
+def _make_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--central-repo",
+                        default="/repo/embargo",
+                        help="The central repository to obtain data from.")
+    parser.add_argument("--image-bucket",
+                        type=str,
+                        default="s3://rubin-summit",
+                        help="The bucket where raw images are, starting with the URI scheme and ``://``")
+    parser.add_argument("--skymap",
+                        type=str,
+                        default="latiss_v1",
+                        help="The skymap name in the central repo.")
+    parser.add_argument("--inst",
+                        default="LATISS",
+                        help="The short instrument name.")
+    parser.add_argument("-d", "--data-query", required=True,
+                        help="The data query for butler to find raw data to process.")
+    parser.add_argument("-n",
+                        default=1,
+                        help="The maximum number of raw datasets to process. Images are "
+                        "selected with the order as given by the iterable from the data "
+                        "query. Up to n images will be processed or until all images from "
+                        "the data query are processed.")
+    return parser
+
+
+def main():
+    args = _make_parser().parse_args()
+    butler = Butler(args.central_repo)
+    inst_name = args.inst
+    local_storage = os.path.join("/lscratch", os.getlogin(), "tmp")
+
+    # Mock up the environment for MiddlewareInterface
+    os.environ["IP_APDB"] = "localhost"
+    os.environ["DB_APDB"] = "postgres"
+    os.environ["USER_APDB"] = "postgres"
+    os.environ["SKYMAP"] = args.skymap
+
+    scheme, image_bucket = args.image_bucket.split("://")
+    interface = MiddlewareInterface(
+        butler, image_bucket, inst_name, os.environ["SKYMAP"], local_storage, scheme + "://"
+    )
+
+    collection = interface.instrument.makeDefaultRawIngestRunName()
+    _log.debug("Use query %s on %s", args.data_query, butler)
+    data_ids = butler.registry.queryDataIds(
+        ["exposure", "detector"],
+        datasets="raw",
+        collections=collection,
+        where=args.data_query,
+    )
+    if not data_ids.count():
+        _log.warning("No raws were found in the data query")
+    _log.debug("Use the first %d of %d in the query results", args.n, data_ids.count())
+    for data_id in data_ids.limit(args.n).expanded():
+        uri = butler.getURI(
+            "raw",
+            collections=collection,
+            exposure=data_id["exposure"],
+            detector=data_id["detector"],
+            instrument=inst_name,
+        )
+        _log.debug("Process %s", data_id.full)
```

Comment:
How long is `data_id.full`? I think just printing `data_id` should give all the information that's actually human-friendly.